src/System.Private.ServiceModel/src/System/ServiceModel/Channels/TextMessageEncoder.cs

```diff
@@ -546,7 +546,9 @@ public override Task<ArraySegment<byte>> WriteMessageAsync(Message message, int
                     await message.WriteMessageAsync(xmlWriter);
                     xmlWriter.WriteEndDocument();
                 }
-                await xmlWriter.FlushAsync();
+
+                // The underlying type of xmlWriter, XmlUTF8TextWriter, has not implemented FlushAsync() yet.
+                xmlWriter.Flush();
```

Comment:
In that case lets open a bug and give a dummy sync implementation which just returns
Task.FromResult();